### PR TITLE
[new_york_pizza_nl_de] Add spider (303 locations)

### PR DESCRIPTION
**Pull request by hfs**
Add [New York Pizza](https://www.newyorkpizza.nl/vestigingen), a pizza chain mostly in NL, but also in DE.

```
2024-11-12 21:57:18 [scrapy.statscollectors] INFO: Dumping Scrapy stats:
{'atp/brand/New York Pizza': 303,
 'atp/brand_wikidata/Q2639128': 303,
 'atp/category/amenity/fast_food': 303,
 'atp/cdn/cloudflare/response_count': 6,
 'atp/cdn/cloudflare/response_status_count/200': 6,
 'atp/field/email/missing': 303,
 'atp/field/image/missing': 303,
 'atp/field/operator/missing': 303,
 'atp/field/operator_wikidata/missing': 303,
 'atp/field/state/missing': 303,
 'atp/field/twitter/missing': 303,
 'atp/item_scraped_host_count/www.newyorkpizza.de': 21,
 'atp/item_scraped_host_count/www.newyorkpizza.nl': 282,
 'atp/nsi/perfect_match': 303,
```

### CODE REVIEW

File: locations/spiders/new_york_pizza_nl_de.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            headers={"RequestVerificationToken": auth_token},
+            callback=self.parse,
+            cb_kwargs={"top_level_domain": top_level_domain},
+        )
+
+    def parse(self, response, top_level_domain=None, **kwargs):
+        for store in response.json():
+            item = DictParser.parse(store)
+            if item["name"].startswith("New York Pizza "):
+                item["branch"] = item["name"].lstrip("New York Pizza ")
```
Comment (Cj-Malone) [inline locations/spiders/new_york_pizza_nl_de.py:38]: strip doesn't do what you think it does. See the serene example https://www.w3schools.com/python/ref_string_lstrip.asp


You want removeprefix

File: locations/spiders/new_york_pizza_nl_de.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+            if item["name"] != "S4D Test":
+                yield item
+
+    @staticmethod
+    def convert_opening_hours(store, language):
+        days_to_en = DAYS_DE if language == "de" else DAYS_EN
+        re_interval = re.compile(r"(\d\d:\d\d) - (\d\d:\d\d)")
+        hours = OpeningHours()
+        for entry in store["opening_hours"]:
+            for weekday in entry["key"].split("-"):
```
Comment (Cj-Malone) [inline locations/spiders/new_york_pizza_nl_de.py:54]: I'm on my phone, so I haven't seen the source data, but this looks wrong. Mo-Fr isn't just Monday and Friday. I think the helper function is called day_range. I can fix this later or tomorrow.

**Comment by hfs** [comment]
Thank you for the thorough review. I added a commit that should fix the issues.

**Comment by Cj-Malone** [comment]
Thank you

**Pull request merged**